### DISCUSSION TITLE
Cursor does not respect moveCursorToEnd when typing new text (on initial load) on iOS

### DISCUSSION THREAD

**Discussion by mark8044** [singerdmx/flutter-quill#715] [Mobile] [answered]
When my editor first opens, I want to pre-fil the editor with some text, in this example: `Test 123`, and then I want to move the cursor to the end using `.moveCursorToEnd();` so that the user is able to immediately start adding their own text after the initial pre-filled text

The cursor properly is properly positioned immediately after `Test 123` but if the user starts typing, the new text that is typed is inserted at the very start of the document, in front of `Test 123` and not after it, where the cursor is currently positioned.

- This problem seems to **occur on iOS**.  For me its on the simulator as I don't have a real device
- This problem **DOES NOT occur on Android** simulator or real devices

```
class TestWidget extends StatefulWidget {
  const TestWidget({Key? key}) : super(key: key);

  @override
  State<TestWidget> createState() => _TestWidgetState();
}

class _TestWidgetState extends State<TestWidget> {

  var _controller = QuillController.basic();
  final FocusNode _focusNode = FocusNode();
  
  @override
  void initState(){
    super.initState();
    _controller.document.insert(0, 'Test123');
    _controller.moveCursorToEnd();
  }


  @override
  Widget build(BuildContext context) {
    return Container(child:
        Column(
          children: [
            QuillToolbar.basic(
            controller: _controller),

            QuillEditor(
              focusNode: _focusNode,
              controller: _controller,
              readOnly: false,
              autoFocus: true,
              padding: const EdgeInsets.fromLTRB(20, 20, 20, 12),
              enableInteractiveSelection: true,
              expands: false,
              scrollable: true,
              scrollController: ScrollController(),
              customStyles: DefaultStyles.getInstance(context).merge(DefaultStyles(
              )),
            ),
            
          ],
        )
      ,);
  }
}
```

**Comment by mark8044** [comment, marked as answer]
Solved it by placing the movecursortoend inside a postframecallback, in case anyone else runs into this issue

**Reply by karroze** [reply]
This absolutely does not solve the issue, unfortunately

**Comment by karroze** [comment]
The issue still persists after 3 years...
